### ISSUE TITLE
as.POSIXct() to Postgres timestamptz round trip does not behave as expected

### ISSUE THREAD

**Issue by mbannert** [r-dbi/RPostgres#250]
Based on how RPostgres reads timestamptz formatted fields, I woulda expected different behavior when writing to the database. The below example runs on a docker container based on the postgrel:latest image. 

The following is correctly returned as CET, which is UTC+1 (January is within winter time). 

```r
as.POSIXct("2020-01-01 12:00:00", tz = "Europe/Zurich")
#> [1] "2020-01-01 12:00:00 CET"
```

When I write this to a PostgreSQL `timestamptz` field

```r
dbExecute(con, "CREATE TEMPORARY TABLE bla(id integer, some_time TIMESTAMPTZ)")
dbExecute(con, "INSERT INTO bla VALUES (1, $1)", as.POSIXct("2020-01-01 12:00:00", tz = "Europe/Zurich"))
dbExecute(con, "INSERT INTO bla VALUES (2, '2020-01-01 12:00:00+01')")
dbGetQuery(con, "SELECT * FROM bla")
```
will return (in R)

id           some_time
 1       2020-01-01 13:00:00
 2      2020-01-01 12:00:00

So what happened? Both inserts end up being stored in the DB's `timestamptz` field as UTC, in other words tz seems to be ignored. That behavior would be somewhat ok though it may waste some time on larger datasets ( it would simply mean I had to convert everything to UTC in R before storing). But what puzzles me is that READING data obviously respects the local time zone and resurrects the time zone... 

In other words, I don't think we should have difference here: 

direct SQL query on a psql client
```SQL
postgres=# SELECT * FROM bla ;
 id |       some_time        
----+------------------------
  1 | 2020-01-01 12:00:00+00
  2 | 2020-01-01 11:00:00+00
(2 rows)
```
vs R

```r

> dbGetQuery(con, "SELECT * FROM bla")
  id           some_time
1  1 2020-01-01 13:00:00
2  2 2020-01-01 12:00:00

```

Is it me or this is a bug / odd 🤷‍♂️  ?

**Comment by mbannert** [comment]
NVM. it seems like this was fixed in 1.2.0 . I had a slightly older version of RPostgres which showed the above behavior. After an update it's gone. 👻
